### PR TITLE
[CARBONDATA-201]add comment option

### PR DESCRIPTION
**Pull request by lion-x**
# Why raise this pr?

add csv comment option to csv paser,and fix one bug when passing quotechar.

# How to solve this?

by default # will as the comment char,and user can use COMMENTCHAR to specify.

# How to test

Pass the exist test cases.

### CODE REVIEW

**Comment by lion-x** [comment]
CI has passed,please check:
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/87/

**Comment by QiangCai** [comment]
@lion-x 
I think it is unnecessary to modify the following files.
CSVFilePartitioner.java
CSVParser.java
CSVReader.java

**Comment by lion-x** [comment]
ok, i will check

File: processing/src/main/java/org/apache/carbondata/processing/csvload/DataGraphExecuter.java [hunk 1]
```diff
@@ -288,6 +288,8 @@ private void processHadoopFileInputMeta(List<String> measureColumns, List<StepMe
             ((CsvInputMeta) step.getStepMetaInterface()).setDelimiter(model.getCsvDelimiter());
             ((CsvInputMeta) step.getStepMetaInterface())
               .setEscapeCharacter(model.getEscapeCharacter());
+            ((CsvInputMeta) step.getStepMetaInterface())
+              .setCommentCharacter(model.getEscapeCharacter());
```
Comment (QiangCai) [inline processing/src/main/java/org/apache/carbondata/processing/csvload/DataGraphExecuter.java]: please check

File: processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/CsvInputMeta.java [hunk 2]
```diff
@@ -162,6 +168,8 @@ private void readData(Node stepnode) throws KettleXMLException {
       blocksID = XMLHandler.getTagValue(stepnode, "blocksID");
       partitionID = XMLHandler.getTagValue(stepnode, "partitionID");
       escapeCharacter = XMLHandler.getTagValue(stepnode, "escapeCharacter");
+      quoteCharacter = XMLHandler.getTagValue(stepnode, "quoteCharacter");
+      commentCharacter = XMLHandler.getTagValue(stepnode, "commentCharacter");
```
Comment (QiangCai) [inline processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/CsvInputMeta.java]: please keep the same sequence:  escapeCharacter -> quoteCharacter -> commentCharacter

File: processing/src/main/java/org/apache/carbondata/processing/csvload/DataGraphExecuter.java [hunk 3]
```diff
@@ -288,6 +288,8 @@ private void processHadoopFileInputMeta(List<String> measureColumns, List<StepMe
             ((CsvInputMeta) step.getStepMetaInterface()).setDelimiter(model.getCsvDelimiter());
             ((CsvInputMeta) step.getStepMetaInterface())
               .setEscapeCharacter(model.getEscapeCharacter());
+            ((CsvInputMeta) step.getStepMetaInterface())
+              .setCommentCharacter(model.getCommentCharacter());
```
Comment (QiangCai) [inline processing/src/main/java/org/apache/carbondata/processing/csvload/DataGraphExecuter.java]: The method "processHadoopFileInputMeta" is useless.

**Comment by lion-x** [comment]
OK, I have already renew some files by your suggestion.

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataWithHiveSyntax.scala [hunk 4]
```diff
@@ -590,6 +590,23 @@ class TestLoadDataWithHiveSyntax extends QueryTest with BeforeAndAfterAll {
...
+      "create table comment_test(imei string, age int, task bigint, num double, level decimal(10," +
+        "3), productdate timestamp, mark int, name string) STORED BY 'org.apache.carbondata.format'"
+    )
+    sql(
+      "LOAD DATA local inpath './src/test/resources/comment.csv' INTO TABLE comment_test " +
+        "options('DELIMITER' = ',', 'QUOTECHAR' = '.', 'COMMENTCHAR' = '?','FILEHEADER'='imei,age,task,num,level,productdate,mark,name')"
+    )
+    checkAnswer(sql("select imei from comment_test"),Seq(Row("\"huawei"),Row("#huawei"), Row(""),
+      Row("~huawei")))
+    sql("drop table if exists comment_test")
```
Comment (QiangCai) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataWithHiveSyntax.scala]: Better to drop table in the method afterAll

Comment (lion-x) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataWithHiveSyntax.scala]: ok

**Comment by QiangCai** [comment]
LGTM

**Comment by lion-x** [comment]
CI has passed,please check:
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/115/

**Comment by lion-x** [comment]
CI has passed,please check:
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/125/

**Pull request closed without merging**